Fix modal style in WordPress 5.8

thomasguillot opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

With WP 5.8 just around the corner, the style for our `Modal` component was broken (using 5.8-RC2-51407).

This PR updates the style to make sure they look OK in 5.8

### How to test the changes in this Pull Request:

1. Make sure you are using WP 5.8
2. Navigate to a page where we have a modal (eg. components demo, campaigns)
3. Open the modal... it should look broken
4. Switch to this branch
5. Refresh page and reopen the modal

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

dkoo reviewed (changes requested):
Huh, not sure what's different between your WP 5.8 and mine, but the modal components did not look broken to me running 5.8-RC2, nor after I updated to 5.8-RC2-51407.

<img width="1270" alt="Screen Shot 2021-07-13 at 10 59 15 AM" src="https://user-images.githubusercontent.com/2230142/125494437-17f979fa-2857-4dd6-a22b-ba73f6975b88.png">
<img width="838" alt="Screen Shot 2021-07-13 at 10 58 54 AM" src="https://user-images.githubusercontent.com/2230142/125494440-b50c3191-f750-4f39-a1b1-b6eb827a4a10.png">

After switching to this branch, the modals look off:

<img width="894" alt="Screen Shot 2021-07-13 at 10 52 28 AM" src="https://user-images.githubusercontent.com/2230142/125494521-34d963a8-1e3e-49df-8145-c862335c936a.png">
<img width="1300" alt="Screen Shot 2021-07-13 at 10 45 04 AM" src="https://user-images.githubusercontent.com/2230142/125494522-b3f5f147-792e-4fa1-a6a0-f9458d1a46ca.png">

thomasguillot commented:
@dkoo I realised I had the Gutenberg plugin active... this is why it looked broken for me! Anyway I fixed the paddings in https://github.com/Automattic/newspack-plugin/pull/1037/commits/3f614dfc23ad5d6067b7ba243d1f8ab19eaf408a

This PR still makes sense since I've simplified some of the paddings/margins and updated the close icon 😊

dkoo reviewed (approved):
Looking better! 👍
